src/api/channel/channel.ts

```diff
@@ -0,0 +1,179 @@
+     * 
+     * This broadcast will be received by all windows that are members of this channel, *except* for the window that
+     * makes the broadcast. This matches the behaviour of the top-level FDC3 `broadcast` function.
+     */
+    public broadcast(context: Context): Promise<void>;
+
+    /**
+     * Returns the last context that was broadcast on this channel. All channels initially have no context, until a 
+     * window is added to the channel and then broadcasts. If there is not yet any context on the channel, this method
+     * will return `null`. The context is also reset back into it's initial context-less state whenever a channel is
+     * cleared of all windows.
+     * 
+     * The context of a channel will be captured regardless of how the context is broadcasted on this channel - whether
+     * using the top-level FDC3 `broadcast` function, or using the channel-level {@link broadcast} function on this 
+     * object.
+     * 
+     * NOTE: Only non-default channels are stateful, for the default channel this method will always return `null`.
+     */
+    public getCurrentContext(): Promise<Context|null>;
+
+    /**
+     * Event that is fired whenever a window broadcasts on this channel.
+     * 
+     * The `channel` property within the event will always be this channel instance.
+     */
+    public addEventListener(event: 'broadcast', listener: (event: {channel: Channel; context: Context}) => void): void;
+
+    /**
+     * Remove this event listener
+     * 
+     */
+    public removeEventListener(event: 'broadcast', listener: (event: {channel: Channel; context: Context}) => void): void;
+}
+
+/**
+ * All windows will start off in this channel.
+ * 
+ * Unlike desktop channels, the default channel has no pre-defined name or visual style. It is up to apps to display 
+ * this in the channel selector as they see fit - it could be as "default", or "none", or by "leaving" a user channel.
+ */
+declare class DefaultChannel extends Channel {
+    type: 'default';
+}
+
+/**
+ * User-facing channels, to display within a colour picker or channel selector component.
+ * 
+ * This list of channels should be considered fixed by applications - the service will own the list of user channels,
+ * making the same list of channels available to all applications, and this list will not change over the lifecycle of
+ * the service.
+ * 
+ * We do not intend to support creation of 'user' channels at runtime, as this then adds considerable complexity when
+ * implementing a channel selector component (must now support events, 'create channel' UI, reflowing of channel 
+ * list, etc).
+ */
+declare class DesktopChannel extends Channel {
+    type: 'desktop';
+
+    /**
+     * A user-readable name for this channel, e.g: `"Red"`
+     */
+    name: string;
+
+    /**
+     * The color that should be associated within this channel when displaying this channel in a UI, e.g: `0xFF0000`.
+     */
+    color: number;
```

Comment:
Do we want to bake the concept of a colour into the underlying FDC3 API? For me this is an implementation concern that sits on top of FDC3. E.g. a particular implementation can extend the base channel type with an extra color property if it needs it.

Reply:
Agreed, also encoding a colour as a number feels ambiguous, as it lacks details of the encoding scheme

Reply:
Agreed, done